app/js/actions/careSetting.js

```diff
@@ -0,0 +1,42 @@
+import axios from 'axios';
+import {
+  INPATIENT_CARESETTING,
+  INPATIENT_CARESETTING_ERROR,
+  OUTPATIENT_CARESETTING,
+  OUTPATIENT_CARESETTING_ERROR,
+} from './actionTypes';
+
+const contextPath = window.location.href.split('/')[3];
+const apiBaseUrl = `/${contextPath}/ws/rest/v1`;
+
+export function fetchInpatientCareSetting() {
+  return dispatch => axios.get(`${apiBaseUrl}/caresetting?q=inpatient`)
+    .then((response) => {
+      dispatch({
+        type: INPATIENT_CARESETTING,
+        inpatientCareSetting: response.data.results[0],
+      });
+    })
+    .catch((error) => {
+      dispatch({
+        type: INPATIENT_CARESETTING_ERROR,
+        error: error.response,
+      });
+    });
+}
+
+export function fetchOutpatientCareSetting() {
+  return dispatch => axios.get(`${apiBaseUrl}/caresetting?q=outpatient`)
+    .then((response) => {
+      dispatch({
+        type: OUTPATIENT_CARESETTING,
+        outpatientCareSetting: response.data.results[0],
+      });
+    })
+    .catch((error) => {
+      dispatch({
+        type: OUTPATIENT_CARESETTING_ERROR,
+        error: error.response,
```

Comment:
If there's a network error, e.g if user is offline, `error.response` would be **undefined**

Reply:
Okay.
If I understand correctly, your recommendation is for me to handle network specific errors also?

Reply:
Yes. If you are displaying an error message or some sort of feedback based on the action dispatched, which I believe you should, then you should handle ensure that `error.response` actually is defined.

Reply:
Alright.
That makes sense.
Will make the necessary changes

Reply:
Have implemented the changes.